pkg/manager/internal.go

```diff
@@ -211,6 +211,12 @@ func (cm *controllerManager) Add(r Runnable) error {
 		cm.nonLeaderElectionRunnables = append(cm.nonLeaderElectionRunnables, r)
 	} else if hasCache, ok := r.(hasCache); ok {
 		cm.caches = append(cm.caches, hasCache)
+		if cm.started {
+			cm.startRunnable(hasCache)
+			if !hasCache.GetCache().WaitForCacheSync(cm.internalCtx) {
```

Comment:
This will never timeout, right? Does that match the standard path, where we start the caches before the `cm` is started?

/ok-to-test

Reply:
It could definitely time out. The remote cluster might not be accessible. Should the cache be started and synced with a different context? If the sync times out, I would then cancel the context. I suppose it also makes more sense to append the cache after the sync completes successfully.

I believe this matches the standard path.  I can't simply call `waitForCache` because it returns immediately if `cm.started` is true. It first iterates through the caches and calls `cm.startRunnable` for each one. Then it iterates through the caches again and calls `cache.GetCache().WaitForCacheSync(ctx)` where `ctx` is `cm.internalCtx`.

Reply:
After looking through the code some more,  and the `WaitForCacheSync` function in `shared_informer.go` in client-go in particular, I am less certain that it will time out.